### DISCUSSION TITLE
Best Practices w/ dim_Customer (1M) set to BOTH or ONE Way to multiple Facts

### DISCUSSION THREAD

**Discussion by TinyTimothy2** [TabularEditor/TabularEditor3#1311] [Data Modeling]
I have 2 Fact tables:
Fact_A
Fact_B
 
I have conforming dimensions
Dim_Date Filter One Way to Fact_A and Fact_B
Dim_User Filter One Way  to Fact_A and Fact_B
Dim_Customer Filter BOTH ways to  to Fact_A and Fact_B

The reason for customer to be BOTH ways by default is because I need to run analysis such as 
Show All customers and all customer attributes even if they do not have a record in FACT_A or Fact_B
 
I have measures that gives count of customers from Fact_A and Fact_B
[Total Customers w/ Fact A]=  DISTINCTCOUNT(Fact_A[Cust_ID])
[Total Customers w/ Fact B]=  DISTINCTCOUNT(Fact_B[Cust_ID])
Measure for Total Customers (can be either in A or B or in for other reasons, future, past, other future Fact Uses.
[Total Customers] = Countrows(dim_Customer)

Pretend I have  
[Total Customers] = 100
[Total Customers w/ Fact A]=20
[Total Customers w/ Fact B]=10
It seems like if I have a table that shows:
CustID, Name, City, Region, [Total Customers], [Total Customers w/ Fact A], [Total Customers w/ Fact B]

I need to have the DEFAULT Dim_Customer Filter BOTH ways to  to Fact_A and Fact_B
 
I am now understanding that is probably not the safest way and perhaps still loading all customers in dim_Customter with 
DEFAULT Dim_Customer Filter One way to  to Fact_A and Fact_B  and use Crossfilter in measures set to BOTH is better option?

My model is very large billion rows in facts and 1 M in customer dim.

Thoughts?

**Comment by mlonsk** [comment]
Hi @TinyTimothy2 
I would advise against having the filter direction set to Both between confirming dimensions and facts. Fundamentally, it can make the model behave in unpredictable ways. 
From the way you describe it, I don't see why your table shouldn't work with one-directional relationships.

Bellow, I have a DAX query that is set up the way you describe your scenario, and in the first line e.g. I can see a customer who does not have orders, but have been invoiced and are counted by the [# Customers] 

![image](https://github.com/TabularEditor/TabularEditor3/assets/11939114/aa487a0d-f635-475c-b94b-b7023822cb2e)

If I misunderstand your question, could you create a demo model to share that shows the issue you are facing?
